updating/understanding-openshift-upgrade-duration.adoc

```diff
@@ -0,0 +1,91 @@
+
+[options="header"]
+|===
+|Component Name |Time Duration (in minutes)
+
+|Cluster Version Operator (CVO)
+|45-120
+
+|Machine Config Operator (MCO)
+|5 per node
+|===
+
+=== Cluster Version Operator target upgrade payload deployment
+The Cluster Version Operator (CVO) unpacks the target upgrade release image and applies to the cluster. This process might take somewhere between 45 to 120 minutes.
+
+=== Machine Config Operator
+The Machine Config Operator (MCO) applies a new machine configuration to each control plane and worker node. During this process, the MCO performs the following tasks on each node of the cluster:
+
+* Node draining
+* OS update staging
+* Node rebooting
+
+The control plane node upgrade duration is more predictable and oftentimes shorter than worker nodes since it does not include large workloads. You can upgrade the worker nodes in parallel by setting the `maxUnavailable` field to greater than 1 in the machine configuration pool. The MCO cordons the number of nodes specified in `maxUnavailable` and marks them unavailable for upgrade.
+
+[IMPORTANT]
+====
+With {product-title} 4 upgrades, an operating system lifecycle management and maintenance is made as a part of the cluster upgrade.  This only applies to {op-system-first} nodes.
+====
+
+The time to complete this process depends on many factors including the node and infrastructure configuration. This process might take 5 minutes to complete per node.
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../architecture/architecture-installation.html#update-service-overview_architecture-installation[About the OpenShift Update Service]
+
+
+== Factors affecting the upgrade duration
+In addition to Cluster Version Operator (CVO) and Machine Config Operator (MCO), the following factors also impact the upgrade duration:
+
+* Pod Disruption Budget (PDB)
+* Cluster node health
+* Number of nodes in the cluster
+* Red Hat Enterprise Linux (RHEL) nodes
+
+=== Pod disruption budget
+You must consider pod disruption budgets and pod anti-affinity when you configure `maxUnavailable` field in the machine configuration pool.
+
+For example, if there is a cluster with three worker nodes, you must configure the disruption budget such that it has minimum two replicas be always running. When you configure pod anti-affinity, you must specify that two replicas should never run on the same node. To avoid pod scheduling issues, you must ensure the availability of minimum two worker nodes.
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../nodes/pods/nodes-pods-configuring.adoc#nodes-pods-configuring-pod-distruption-about[pod disruption budget]
+* xref:../nodes/scheduling/nodes-scheduler-pod-affinity.adoc#nodes-scheduler-pod-affinity-about[pod anti-affinity]
+
+=== Cluster node health
+An unhealthy node takes longer time to upgrade. It is recommended to resolve any issues before you begin the cluster upgrade.
```

Comment:
and an unhealthy node (per the MCO's definition of "ready") will continue to hold a `maxUnavailable` slot for the pool.  And we aren't suggesting identification or mitigation steps here to help the admin "resolve any issues" or know if they have resolved the issues or not.  Probably involve the MCO developers in details, but ideally we'd be alerting on any issues that needed resolution, so maybe we can just suggest folks deal with any firing alerts?

Reply:
@wking Lala has suggested a rewrite for this sentence:

> === Cluster node health
> Before you begin the upgrade, you must ensure that all the nodes are available. Any unavailable nodes can significantly increase the upgrade duration because the node unavailability affects the maxUnavailable and pod disruption budgets.
> 
> To check the status of nodes, run the following command:
> 
> $ oc get nodes -o wid

wdyt?